documentation/v0_9_0.md

```diff
@@ -0,0 +1,368 @@
+The feature currently works for mixed assembly of co-dimension 0 (sub-set of cells) and co-dimension 1 (sub-set of facets).
+
+An example of usage is shown below, where a `TrialFunction` from a parent mesh is combined in a variational form with a `TestFunction` from a sub-mesh of all
+exterior facets. We use the integration `ds` over the parent mesh, to ensure that we can use quantities such as `ufl.FacetNormal`.
+
+```python
+domain = dolfinx.mesh.create_unit_square(MPI.COMM_WORLD, 10, 10)
+V = dolfinx.fem.functionspace(domain, ("Lagrange", 1))
+
+# Create submesh and function space on submesh
+tdim = domain.topology.dim
+domain.topology.create_connectivity(tdim - 1, tdim)
+facet_domain, sub_to_parent, _, _ = dolfinx.mesh.create_submesh(
+	domain, tdim - 1, dolfinx.mesh.exterior_facet_indices(domain.topology)
+)
+Q = dolfinx.fem.functionspace(domain, ("DG", 2))
+u = ufl.TrialFunction(V)
+q = ufl.TestFunction(Q)
+a_mixed = ufl.inner(u, q) * ufl.ds(domain=domain)
+
+# Invert entity map to go from integration domain to sub mesh
+facet_imap = msh.topology.index_map(tdim - 1)
+num_facets = facet_imap.size_local + facet_imap.num_ghosts
+parent_to_sub = np.full(num_facets, -1)
+parent_to_sub[sub_to_parent] = np.arange(len(sub_to_parent))
+entity_maps = {facet_domain:parent_to_sub}
+
+# Create dolfinx.fem.Form
+a_compiled = dolfinx.fem.form(a_mixed)
+```
+
+#### Relevant pull requests since last release:
+
+- Codim-1 mixed assembly: [DOLFINx \#3224](https://github.com/FEniCS/dolfinx/pull/3224)
+- Interior facet assembly with submesh of codimensional 1: [DOLFINx \#3452](https://github.com/FEniCS/dolfinx/pull/3452)
+- Submesh of vertices: [DOLFINx \#3455](https://github.com/FEniCS/dolfinx/pull/3455)
+- C++ demo: [DOLFINx \#3436](https://github.com/FEniCS/dolfinx/pull/3436), [DOLFINx \#3262](https://github.com/FEniCS/dolfinx/pull/3262)
+- Packing of coefficients for more than 2 sub-meshes: [DOLFINx \#3260](https://github.com/FEniCS/dolfinx/pull/3260), [DOLFINx \#3361](https://github.com/FEniCS/dolfinx/pull/3361)
+- Fix for block vector assembly: [DOLFINx \#3346](https://github.com/FEniCS/dolfinx/pull/3346)
+- Add create submesh documentation in Python: [DOLFINx \#3112](https://github.com/FEniCS/dolfinx/pull/3112)
+- Consistent orientation of sub-entities, [DOLFINx \#3209](https://github.com/FEniCS/dolfinx/pull/3209)
+- Interpolation to and from co-dim 0 meshes, [DOLFINx \#3114](https://github.com/FEniCS/dolfinx/pull/3114)
+
+---
+
+## Simpler construction of block forms {#extract-blocks}
+
+**Main contributor:** [Jørgen S. Dokken](https://github.com/jorgensd/)
+
+Related to the introduction of [mixed dimensional assembly](#mixed-assembly), having an easier way of creating blocked variational forms is crucial for the usability of the feature. This feature can also be used for "standard" block forms, used with `dolfinx.fem.petsc.assemble_*_block/nest`.
```

Comment:
This says very little - what is it and what problem does it solve?

Reply:
It solves the problem that using the blocked interface/nest interface of PETSC required the user to write out the blocks of the matrix (potentially jacobian of residual) and the residual by hand, making the code super verbose, but also error prone when doing differentiation.

as you can see from the example below and the modified demos in DOLFINx it makes it much easier to code up variational forms.